File: ruby/pangram/pangram.rb
module Pangram
  NON_ASCII_LETTERS = /[^a-z]/.freeze
  private_constant :NON_ASCII_LETTERS
  NUMBER_OF_LETTERS_IN_ALPHABET = 26
  private_constant :NUMBER_OF_LETTERS_IN_ALPHABET

  module_function

  def pangram?(phrase)
    letter_count(phrase) == NUMBER_OF_LETTERS_IN_ALPHABET
  end

  def letter_count(phrase)
    phrase.downcase.gsub(NON_ASCII_LETTERS, "").chars.uniq.count
  end
  private_class_method :letter_count
end

File: javascript/pangram/pangram.js
export const isPangram = phrase => {
  return letterCount(phrase) === NUMBER_OF_LETTERS_IN_ALPHABET
}

function letterCount(phrase) {
  const letters = phrase
    .toLowerCase()
    .replace(NON_ASCII_LETTERS, "")
    .split("")
  return [...new Set(letters)].length
}

const NON_ASCII_LETTERS = /[^a-z]/g
const NUMBER_OF_LETTERS_IN_ALPHABET = 26
